docs/source/llms/dspy/notebooks/dspy_quickstart.ipynb

```diff
@@ -0,0 +1,1265 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "fa602d14-6372-4e0a-92d8-6f7db048c3b8",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "[DSPy](https://dspy-docs.vercel.app/) simplifies building language model (LM) pipelines by replacing manual prompt engineering with structured \"text transformation graphs.\" These graphs use flexible, learning modules that automate and optimize LM tasks like reasoning, retrieval, and answering complex questions. \n",
+    "\n",
+    "### How does it work?\n",
+    "At a high level, DSPy optimizes prompts, selects the best language model, and can even fine-tune the model using training data.\n",
+    "\n",
+    "The process follows these three steps, common to most DSPy [optimizers](https://dspy-docs.vercel.app/docs/building-blocks/optimizers):\n",
+    "\n",
+    "1. **Candidate Generation**: DSPy finds all `Predict` modules in the program and generates variations of instructions and demonstrations (e.g., examples for prompts). This step creates a set of possible candidates for the next stage.\n",
```

Comment:
"instructions and demonstrations" => "instructions or demonstrations", as many optimizers won't touch "instructions".